src/test/java/com/client/core/base/tools/entitychanger/impl/StandardEntityChangerTest.java

```diff
@@ -259,14 +260,35 @@ public void setValueAddsUnknownPropertiesToAdditionalPropertiesIfPossible() {
         assertEquals(10, ((Map) candidate.getAdditionalProperties().get("missingRelation")).get("id"));
     }
 
+    @Test
+    public void setFieldCastsIntegerToString() {
+        EntityChanger entityChanger = new StandardEntityChanger();
+        entityChanger.setField(candidate, "customText1", 10);
+        assertEquals(candidate.getCustomText1(), "10");
+    }
+
+    @Test
+    public void setFieldCastsDoubleToString() {
+        EntityChanger entityChanger = new StandardEntityChanger();
+        entityChanger.setField(candidate, "customText1", 10.5d);
+        assertEquals(candidate.getCustomText1(), "10.5");
+    }
+
+    @Test
+    public void setFieldSavesFailedCastsToAdditionalPropertiesIfAvailable() {
+        EntityChanger entityChanger = new StandardEntityChanger();
+        entityChanger.setField(testBullhornEntity, "simpleStringContainer", "Test");
+        assertEquals("Test", testBullhornEntity.getAdditionalProperties().get("simpleStringContainer"));
+    }
+
```

Comment:
Could we add a test like the following?
```
@Test
    public void setFieldSavesFailedCastsToAdditionalPropertiesIfAvailable() {
        EntityChanger entityChanger = new StandardEntityChanger();
        entityChanger.setField(testBullhornEntity, "simpleStringContainer.someOtherContainer.id", "Test");
        assertEquals("Test", testBullhornEntity.getAdditionalProperties().get("simpleStringContainer.someOtherContainer.id"));
    }
```

I think this will actually throw an error in our case